src/tool/xmeta/xmeta_emit.cpp

```diff
@@ -490,42 +490,32 @@ namespace xlang::xmeta
         ////    &token_event);
     }
     
-    void xmeta_emit::listen_struct_model(std::shared_ptr<struct_model> const& /*model*/) 
+    void xmeta_emit::listen_struct_model(std::shared_ptr<struct_model> const& model) 
     {
-        //auto const& type_name = model->get_id();
-        //static constexpr DWORD struct_type_flag = tdPublic | tdSealed | tdClass | tdSequentialLayout | tdWindowsRuntime;
-        //mdTypeDef implements[] = { mdTokenNil };
-        //auto token_struct_type_def = define_type_def(type_name, struct_type_flag, token_enum, implements);
-
-        //auto iter = type_references.find(type_name);
-        //if (iter == type_references.end())
-        //{
-        //    throw_invalid("Failed to find TypeRef for: " + type_name);
-        //}
-        //TypeRef const& struct_type_ref = iter->second;
-
-        //static constexpr DWORD enumerator_flag = fdHasDefault | fdLiteral | fdStatic | fdPublic;
-        //signature_blob enumerator_signature;
-        //enumerator_signature.add_signature(FieldSig{ TypeSig{struct_type_ref.coded_index<TypeDefOrRef>()} });
+        auto const& struct_name = model->get_fully_qualified_id();
+        static constexpr DWORD struct_type_flag = tdPublic | tdSealed | tdClass | tdSequentialLayout | tdWindowsRuntime;
```

Comment:
Leave a 'future' comment in here that the winrt bit usage may change in the future since xlang & winrt type systems are distinct.

Reply:
Will add in later PR.